main.go

```diff
@@ -1,452 +1,111 @@
-		currentDir, err := pathutil.CurrentWorkingDirectoryAbsolutePath()
-		if err != nil {
-			failf("Failed to get current dir, error: %s", err)
-		}
-		workDir = currentDir
-	} else {
-		absWorkDir, err := pathutil.AbsPath(workDir)
-		if err != nil {
-			failf("Failed to expand path (%s), error: %s", workDir, err)
-		}
-		workDir = absWorkDir
-	}
-
-	log.Donef("Expanded WorkDir: %s", workDir)
-
-	cmdLocator := env.NewCommandLocator()
-	factory, err := ruby.NewCommandFactory(command.NewFactory(env.NewRepository()), cmdLocator)
-	if err != nil {
-		panic(err)
-	}
-
-	if _, err := cmdLocator.LookPath("rbenv"); err != nil {
-		cmd := factory.Create("rbenv", []string{"versions"}, &command.Opts{
-			Stderr: os.Stderr,
-			Stdout: os.Stdout,
-			Dir:    workDir,
-		})
-
-		fmt.Println()
-		log.Donef("$ %s", cmd.PrintableCommandArgs())
-		if err := cmd.Run(); err != nil {
-			log.Warnf("%s", err)
-		}
-	}
-
-	// Select and fetch Apple authentication source
-	var devportalConnectionProvider *devportalservice.BitriseClient
-	if config.BuildURL != "" && config.BuildAPIToken != "" {
-		devportalConnectionProvider = devportalservice.NewBitriseClient(retry.NewHTTPClient().StandardClient(), config.BuildURL, string(config.BuildAPIToken))
-	} else {
-		fmt.Println()
-		log.Warnf("Connected Apple Developer Portal Account not found. Step is not running on bitrise.io: BITRISE_BUILD_URL and BITRISE_BUILD_API_TOKEN envs are not set")
-	}
-	var conn *devportalservice.AppleDeveloperConnection
-	if config.BitriseConnection != "off" && devportalConnectionProvider != nil {
-		var err error
-		conn, err = devportalConnectionProvider.GetAppleDeveloperConnection()
-		if err != nil {
-			handleSessionDataError(err)
-		}
-	}
+	os.Exit(run())
+}
 
-	authConfig, err := appleauth.Select(conn, authSources, authInputs)
-	if err != nil {
-		if _, ok := err.(*appleauth.MissingAuthConfigError); !ok {
-			failf("Could not configure Apple Service authentication: %v", err)
-		}
-		fmt.Println()
-		log.Warnf("No authentication data found matching the selected Apple Service authentication method (%s).", config.BitriseConnection)
-		if conn != nil && (conn.APIKeyConnection == nil && conn.AppleIDConnection == nil) {
-			fmt.Println()
-			log.Warnf("%s", notConnected)
-		}
-	}
+func run() int {
+	logger := log.NewLogger()
+	buildStep := createStep(logger)
 
-	// Split lane option
-	laneOptions, err := shellquote.Split(config.Lane)
+	config, err := buildStep.ProcessConfig()
 	if err != nil {
-		failf("Failed to parse lane (%s), error: %s", config.Lane, err)
+		buildStep.logger.Println()
+		buildStep.logger.Errorf(formattedError(fmt.Errorf("Failed to process Step inputs: %w", err)))
+		return 1
 	}
 
 	// Determine desired Fastlane version
 	fmt.Println()
-	log.Infof("Determine desired Fastlane version")
-
-	gemVersions, err := parseGemfileLock(workDir)
+	buildStep.logger.Infof("Determine desired Fastlane version")
+	gemVersions, err := parseGemfileLock(config.WorkDir)
```

Comment:
I think this should go to the `ProcessConfig` method.

Reply:
Now that I moved this code to `ProcessConfig` and looked at it a little more closely I realized if we don’t find a fastlane version in their gemfile we just print an error. Do we also need to fail at this point (e.g: return nil config and the error we get) or is it ok to continue execution?

Reply:
I would keep the original behavior of `parseGemfileLock` and how we handle the error of `parseGemfileLock`.

Reply:
d7a7e0130b528e394be915f0e551c33e66079917